### PR TITLE
Consolidate stress exit and log flushing

### PR DESCRIPTION
**Pull request by ChumpChief**
This PR consolidates the process exit paths for the stress test, as well as the log flush paths.  This closes some gaps where we would exit the process without flushing the logs (e.g. if initialize() threw), fixes some cases where we would exit with status 0 despite error (e.g. any error in the orchestrator), as well as makes it easier to see and audit the exit and flush flows.

Summary of notable changes:
1. FileLogger now takes in the output directory and file prefix in its constructor, rather than deciding on them at flush time.  This enforces that the logger is fully set up early and always ready to flush.  Previously, we had no real way to flush prior to container attach.
2. There is now a single logger instance per process, and injected into whatever functionality needs it (rather than the current pseudo-singleton with multiple instances).  This is nice for simplification, but is also needed so we can consolidate the flush() call high on the stack.
3. FileLogger now always outputs logs, rather than suppressing logs without errors.  Logs are useful even in non-error cases.
4. The try/catch/error code strategy is now mirrored between the orchestrator and runners, whereas they were subtly different before.
5. Ensure the base logger is flushed before trying to write the file.  The file write block is synchronous anyway (no opportunity for parallelism) and if it errors then the promise floats and might get unceremoniously interrupted by process exit if the file write fails.
6. General error/exit strategy is to let all errors throw (rather than try to make a local decision to exit the process), and then catch them high on the stack for flush and process exit.

### CODE REVIEW

File: packages/test/test-service-load/src/FileLogger.ts [hunk 1]
```diff
@@ -3,80 +3,85 @@
...
+// This test expects that if a certain env variable is specified, it points to a package that will
+// pollute the global with a "getTestLogger" when imported.  getTestLogger is actually expected to be
+// "instantiateTestLogger" in practice (it creates a new one, rather than retrieving an existing one).
+// Generally speaking, this global logger type will be the one that actually knows how to log to a real
+// destination (i.e. aria-logger).
+// TODO: Consider injecting a logger rather than relying on an environment variable and dynamic import.
+// TODO: Consider just exporting the function and importing it directly rather than polluting the global.
+const maybeInstantiateGlobalLoggerType = async () => {
+	if (process.env.FLUID_TEST_LOGGER_PKG_SPECIFIER !== undefined) {
+		if (getTestLogger === undefined) {
```
Comment (alexvy86) [inline packages/test/test-service-load/src/FileLogger.ts:24]: I was a bit confused by the refactor but finally got what it's doing. I'd suggest an explicit comment noting that checking `getTestLogger` seemingly before the `await import` that would define it is to optimize repeated calls to `maybeInstantiateGlobalLoggerType` (even if we don't technically do that).

**Review by alexvy86 — APPROVED** [review]
Overall lgtm

File: packages/test/test-service-load/src/FileLogger.ts (hunk 1 above)
Comment (ChumpChief) [reply packages/test/test-service-load/src/FileLogger.ts:26]: Added a comment to that effect in latest.  Realistically it should be benign to double import here as it should just reassign the same function to getTestLogger, but it felt silly :)

**Review by anthony-murphy — APPROVED** [review]

File: packages/test/test-service-load/src/stressTest.ts [hunk 2]
```diff
@@ -113,42 +122,36 @@ export async function stressTest(
 		}, 20000);
 	}
 
-	try {
```
Comment (markfields) [inline packages/test/test-service-load/src/stressTest.ts:116]: @ChumpChief  why did you get rid of this try-finally?  If the callback to `runnerArgs.map` throws, the ending logs won't be logged to console.

Comment (ChumpChief) [reply packages/test/test-service-load/src/stressTest.ts:116]: The throw should propagate up to the caller (`main()`) - there I have a try/finally where it should be caught and logged (well, at least the first failure will - the handling still isn't great in the case of multiple failures).  Are you seeing cases where errors don't get caught there?

Comment (ChumpChief) [reply packages/test/test-service-load/src/stressTest.ts:116]: Followed up with Mark - this comment was referring to the logging of "end time" and "run time" at the end of the test.  I omitted these in the error case intentionally, as the test didn't successfully complete and so the "end time" is somewhat nonsensical.

Fair to make changes here going forward though - maybe with additionally clarifying the log text to explain if its referring to a successful run duration vs. how long it ran before erroring out.

**Pull request merged**